core/config.go

```diff
@@ -111,3 +132,52 @@ func (cc ChainProverConfig) Build() (*ProvableChain, error) {
 	}
 	return NewProvableChain(chain, prover), nil
 }
+
+func SyncChainConfigsFromEvents(pathName string, msgIDsSrc, msgIDsDst []MsgID, src, dst *ProvableChain, configID ConfigIDType) error {
+	if err := SyncChainConfigFromEvents(pathName, msgIDsSrc, src, configID); err != nil {
+		return err
+	}
+	if err := SyncChainConfigFromEvents(pathName, msgIDsDst, dst, configID); err != nil {
+		return err
+	}
+	return nil
+}
+
+func SyncChainConfigFromEvents(pathName string, msgIDs []MsgID, chain *ProvableChain, configID ConfigIDType) error {
+	for _, msgID := range msgIDs {
+		if msgID == nil {
+			continue
+		}
+		msgRes, err := chain.Chain.GetMsgResult(msgID)
+		if err != nil {
+			return fmt.Errorf("failed to get message result: %v", err)
+		} else if ok, failureReason := msgRes.Status(); !ok {
+			return fmt.Errorf("msg(id=%v) execution failed: %v", msgID, failureReason)
+		}
+
+		for _, event := range msgRes.Events() {
+			var id string
+			switch configID {
+			case ConfigIDClient:
+				if clientIdentifier, ok := event.(*EventGenerateClientIdentifier); ok {
+					id = clientIdentifier.ID
+				}
+			case ConfigIDConnection:
+				if connectionIdentifier, ok := event.(*EventGenerateConnectionIdentifier); ok {
+					id = connectionIdentifier.ID
+				}
+			case ConfigIDChannel:
+				if channelIdentifier, ok := event.(*EventGenerateChannelIdentifier); ok {
+					id = channelIdentifier.ID
+				}
+			}
```

Comment:
It is unnecessary to give `configID` as a parameter to `SyncChainConfigFromEvents`.
See the following code.

```
var configID ConfigIDType
switch event := event.(type) {
case *EventGenerateClientIdentifier:
    configID = ConfigIDClient
    ...
case *EventGenerateConnectionIdentifier:
    configID = ConfigIDConnection
    ...
case *EventGenerateChannelIdentifier:
    configID = ConfigIDChannel
    ...
}
```

Reply:
https://github.com/hyperledger-labs/yui-relayer/commit/286bfe5359804373830a090058994c3dbbb0447b
Removed configID parameter